src/interval.jl

```diff
@@ -74,6 +74,44 @@ Interval(f, l, x::Bool, y::Bool) = Interval(f, l, Inclusivity(x, y))
 Interval(f::T, l::T) where T = Interval(f, l, Inclusivity(true, true))
 (..)(f::T, l::T) where T = Interval(f, l)
 
+"""
+    merge(intervals::T) where T <: AbstractVector{<:AbstractInterval}
+
+Flattens a vector of overlapping intervals into smaller vector containing only
+non-overlapping intervals.
+"""
+function merge(intervals::AbstractVector{<:AbstractInterval{T}}) where T
+    length(intervals) < 2 && return intervals
+
+    sort!(intervals)
```

Comment:
Note that this will mutate the argument passed to `merge`, so by convention the function should be named `merge!`. If it's not the intent to mutate the input, I recommend using `sort` instead of `sort!`, which will make a copy.

It may be worthwhile to go all-in on the in-place approach and overwrite the input completely during the computation and return the mutated original array. Then the out-of-place version could be defined as `merge(intervals) = merge!(copy(intervals))`. Just a suggestion.